src/components/Tokens/TokenDetails/TokenDescription.tsx

```diff
@@ -0,0 +1,156 @@
+import { Trans } from '@lingui/macro'
+import { ChainId } from '@uniswap/sdk-core'
+import Column from 'components/Column'
+import { EtherscanLogo } from 'components/Icons/Etherscan'
+import { Globe } from 'components/Icons/Globe'
+import { TwitterXLogo } from 'components/Icons/TwitterX'
+import CurrencyLogo from 'components/Logo/CurrencyLogo'
+import Row from 'components/Row'
+import { NoInfoAvailable, truncateDescription, TruncateDescriptionButton } from 'components/Tokens/TokenDetails/shared'
+import { useTokenProjectQuery } from 'graphql/data/__generated__/types-and-hooks'
+import { chainIdToBackendName } from 'graphql/data/util'
+import { useCurrency } from 'hooks/Tokens'
+import { useColor } from 'hooks/useColor'
+import useCopyClipboard from 'hooks/useCopyClipboard'
+import { useCallback, useReducer } from 'react'
+import { Copy } from 'react-feather'
+import styled from 'styled-components'
+import { BREAKPOINTS } from 'theme'
+import { ClickableStyle, EllipsisStyle, ExternalLink, ThemedText } from 'theme/components'
+import { opacify } from 'theme/utils'
+import { shortenAddress } from 'utils'
+import { ExplorerDataType, getExplorerLink } from 'utils/getExplorerLink'
+
+const TokenInfoSection = styled(Column)`
+  gap: 12px;
+  width: 100%;
+
+  @media (max-width: ${BREAKPOINTS.lg - 1}px) and (min-width: ${BREAKPOINTS.sm}px) {
+    max-width: 45%;
+  }
+`
+
+const TokenNameRow = styled(Row)`
+  gap: 8px;
+  width: 100%;
+`
+
+const TokenName = styled(ThemedText.BodyPrimary)`
+  ${EllipsisStyle}
+`
+
+const TokenButtonRow = styled(TokenNameRow)`
+  flex-wrap: wrap;
+`
+
+const TokenInfoButton = styled(Row)<{ tokenColor: string }>`
+  gap: 8px;
+  padding: 8px 12px;
+  border-radius: 20px;
+  color: ${({ tokenColor }) => tokenColor};
+  background-color: ${({ tokenColor }) => opacify(12, tokenColor)};
+  font-size: 14px;
+  font-weight: 535;
+  line-height: 16px;
+  width: max-content;
+  ${ClickableStyle}
+`
+
+const TokenDescriptionContainer = styled(ThemedText.BodyPrimary)`
+  ${EllipsisStyle}
+  max-width: 100%;
+  max-height: fit-content;
+  line-height: 24px;
+  white-space: pre-wrap;
+`
+
+const TRUNCATE_CHARACTER_COUNT = 75
+
+export function TokenDescription({
+  tokenAddress,
+  chainId = ChainId.MAINNET,
+  showCopy = true,
```

Comment:
Don't think it's necessary to default to `true`. Then we can use the implicit truth value of JSX props.